privaterelay/signals.py

```diff
@@ -17,10 +19,23 @@ def invitations_only(sender, **kwargs):
     for allowed_domain in ALLOWED_SIGNUP_DOMAINS:
         if domain_part == allowed_domain:
             return True
+
+    active_accounts_count = User.objects.count()
+    if active_accounts_count >= settings.MAX_ACTIVE_ACCOUNTS:
+        raise PermissionDenied
+
+    if kwargs['request'].session['alpha_token'] == settings.ALPHA_INVITE_TOKEN:
```

Comment:
Being nit-picky, but I am sure there will be people who were invited but did not visit `/invitation` before they tried to login and use the service resulting to a error page with stack trace for `alpha_token` key not found in the session object. My recommendation is have an additional check for the `alpha_token` value in the session to throw a helpful message in the permission denied view like:
```python
if not kwargs['request'].session.has_key('alpha_token'):
    raise PermissionDenied('You must visit the invitation link provided in your invite email before attempting to participate in testing of Private Relay.')
```
(Side note, tried this out and `PermissionDenied` exception does not display the message because I think something is catching the exception and displaying the `authentication_error.html`. If we can pass a context value for the customized message to redirect the user to the correct flow that would be great.)

Reply:
I updated and cleaned up the logic in the signal handler.

I tried a bit to see if we could return or raise a better error message for the page, but that turned into a bit of a rabbit-hole.

If we want to improve it for the invite-only beta, I think we'll use [messages](https://docs.djangoproject.com/en/2.2/ref/contrib/messages/) instead. But for now, alpha users may just get not-the-best messages.